packages/salesforce/test/Adaptor.test.js

```diff
@@ -194,4 +195,138 @@ describe('Adaptor', () => {
       expect(result).to.eql('Nhamaonha 6a Classe 2023-10-09');
     });
   });
+
+  describe('query', () => {
+    it('should properly pass the query string', async () => {
+      const fakeConnection = {
+        query: function () {
+          return Promise.resolve({
+            done: true,
+            totalSize: 1,
+            records: [{ Name: 'OpenFn' }],
+          });
+        },
+      };
+
+      let state = {
+        connection: fakeConnection,
+        references: [],
+      };
+      let spy = sinon.spy(fakeConnection, 'query');
+
+      query('select Name from Account')(state).then(state => {
+        expect(spy.calledWith('select Name from Account')).to.eql(true);
+      });
+    });
+    it('should properly pass the query string from state', async () => {
+      const fakeConnection = {
+        query: function () {
+          return Promise.resolve({
+            done: true,
+            totalSize: 1,
+            records: [{ Name: 'OpenFn' }],
+          });
+        },
+      };
+
+      let state = {
+        connection: fakeConnection,
+        references: [],
+        qs: 'select Name from Account',
+      };
+      let spy = sinon.spy(fakeConnection, 'query');
+
+      query(state => state.qs)(state).then(state => {
+        expect(spy.calledWith('select Name from Account')).to.eql(true);
+      });
+    });
+
+    it('should fetch up to 2000 records', done => {
```

Comment:
I'm still not happy about this test :thinking: 

Based on your last comment in the other thread, I still don't think you have a clear vision of what you're trying to test here.

I know I'm pushing you hard on this but I really think we need to raise our test standards across our core adaptors in particular. I want 2 or 3 really good, sharp, focused, valuable tests. I want military precision. I don't want 10 or 20 vague, inconsistent and ineffective ones.

Let's forget about that for a moment: this test says `should fetch up to 2000 records`, but the mock returns only 1 record. So it isn't testing 2000 records at all.

I would be happy enough if you renamed the test to something like `should return some records` and we treat it as a general happy-path test of the function.

Alternatively, you could rename the test to `should fetch 1 record` and at least it's accurate.

Reply:
I am happy with `should fetch 1 record`. It's a bit hard to create fixtures for more than `2000` records. Perhaps this is will require factories library like chancejs or [factory-girl](https://github.com/simonexmachina/factory-girl)

Reply:
Well it's super easy to create a 2000 record test , just do `new Array(2000).fill({})`. The contents of the array are totally immaterial.

But fetch 1 is absolutely fine